app/src/main/java/com/example/memorieskeeper/LoginActivity.java

```diff
@@ -0,0 +1,90 @@
+
+import com.example.memorieskeeper.services.CustomGoogleAuth;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.GoogleAuthProvider;
+
+public class LoginActivity extends AppCompatActivity {
+    private static final String TAG = "GoogleActivity";
+
+    ActivityResultLauncher<Intent> signInLauncher;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        signInLauncher = registerForActivityResult(
+                new ActivityResultContracts.StartActivityForResult(),
+                result -> {
+                    if (result.getResultCode() == Activity.RESULT_OK) {
+                        Intent data = result.getData();
+                        Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+                        handleSignInResult(task);
+                    }
+                });
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(this);
+        updateUI(account != null);
+    }
+
+    private void signIn() {
+        Intent signInIntent = CustomGoogleAuth.getGoogleSignInClient(this).getSignInIntent();
+        signInLauncher.launch(signInIntent);
+    }
+
+    private void handleSignInResult(Task<GoogleSignInAccount> completedTask) {
+        try {
+            GoogleSignInAccount account = completedTask.getResult(ApiException.class);
+            if (account != null) {
+                firebaseAuthWithGoogle(account.getIdToken());
+            }
+        } catch (ApiException e) {
+            Log.w(TAG, "signInResult:failed code=" + e.getStatusCode());
+            updateUI(false);
+        }
+    }
+
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
+        FirebaseAuth.getInstance().signInWithCredential(credential)
+                .addOnCompleteListener(this, task -> {
+                    if (task.isSuccessful()) {
+                        updateUI(true);
+                    } else {
+                        Log.w(TAG, "signInWithCredential:failure", task.getException());
+                    }
+                });
+    }
+
+    private void updateUI(boolean isSignedIn) {
+        setContentView(R.layout.activity_login);
+
+        findViewById(R.id.btnSignIn).setOnClickListener(view -> signIn());
+
+        if (isSignedIn) {
+            startActivity(new Intent(this, HomeActivity.class));
+        }
+    }
```

Comment:
Let's move `setOnClickListener` to `onCreate` method and inline `startActivity(new Intent(this, HomeActivity.class));` where it is needed.